GitUI/Hotkey/HotkeySettingsManager.cs

```diff
@@ -159,9 +169,9 @@ void FillDictionaryWithCommands()
             string CalcDictionaryKey(string settingName, int commandCode) => settingName + ":" + commandCode;
         }
 
-        private static HotkeySettings[]? LoadSerializedSettings()
+        private static HotkeySettings[]? LoadSerializedSettings(HotkeySettingsManager hotkeySettingsManager)
```

Comment:
Either
```suggestion
        private HotkeySettings[]? LoadSerializedSettings()
```
or
```suggestion
        private static HotkeySettings[]? LoadSerializedSettings(Action<HotkeySettings[]?> saveSettings)
```

Reply:
Why? This would bloat the code... What's the benefit?

Reply:
Isn't the following kind of weird?

```
class C
{
    public void foo()
    {
        bar(this);
    }

    private static void bar(C c)
    {
        c.baz();
    }

    private void baz()
    {
    }
}
```

Reply:
In your example it may look strange. However, the actual implementation is a bit more intricate.

I think that you're maybe overthinking this, these are all private methods.

Reply:
> I think that you're maybe overthinking this

No, I wonder what you want to express. This is an instance method, isn't it? Why explicitly specify the `this` pointer as argument? 
If you want to emphasize that these two pseudo-static methods call `HotkeySettingsManager.SaveSettings()` only, then rather as specific argument `Action<HotkeySettings[]?> saveSettings`.

![image](https://github.com/gitextensions/gitextensions/assets/36601201/99d50e09-d656-4f9e-8f5f-4af6ba51a2eb)

Reply:
Passing `this` into static is more efficient than creating an `Action` object - "Code size 9 (0x9)" vs "Code size 20 (0x14)" ([static](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQyDMABGgQMIEDeOB1RhyALAQLIAUAlBVTdwMovAALMAGc2Abi7UAvpIKyADgCcwANwCGwAKZFGAEQD27CgRnZus/EXQA2HQT5kAxh0pnuNRwDoD7CW+k4pjhAA) vs [Action](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQPkGYACNQgYXUIG8dDbijkAWQgWQAoBKKmu3gZTbAAFmADOAOgAiAew4BuHrQC+iwqoAOAJzAA3AIbAApsWYzOVQitzZeBYugBsJwgOTpUhAMZdqN3nU9OBT9lHCUgA===))

Other than that, this is a shit code anyway, and IMO we're spending too much time discussing this.